### PR TITLE
TAJO-1265: min(), max() does not handle null properly

### PR DESCRIPTION
**Pull request by sirpkt**
min() and max() handle null value separately from non-null values.
It computes min() or max() among non-null values first.
If no non-null value exists, they return null value.
Implementation style of min() and max() is also changed that 
there are abstract classes of Min and Max common for all the data types
and only type specific methods are implemented for each type.

### CODE REVIEW

**Comment by sirpkt** [comment]
I also modified sum() and avg() to handle null properly.

**Comment by hyunsik** [comment]
I'll review it tonight. Thank you for your contribution.

**Comment by hyunsik** [comment]
The patch looks good to me. It corrects the wrong implementation. There are some trivial things that I need to check. I'll finish the review by tomorrow.

File: tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/AvgDouble.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 package org.apache.tajo.engine.function.builtin;
 
+import org.apache.commons.lang.ObjectUtils;
```
Comment (hyunsik) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/AvgDouble.java]: Unused import

File: tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/AvgLong.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 package org.apache.tajo.engine.function.builtin;
 
+import org.apache.commons.lang.ObjectUtils;
```
Comment (hyunsik) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/AvgLong.java]: unused import

File: tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/SumDouble.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
 import org.apache.tajo.datum.Datum;
 import org.apache.tajo.datum.DatumFactory;
 import org.apache.tajo.datum.Float8Datum;
```
Comment (hyunsik) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/SumDouble.java]: unused import

File: tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/SumLong.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 
 package org.apache.tajo.engine.function.builtin;
 
+import org.apache.commons.math.stat.descriptive.summary.Sum;
```
Comment (hyunsik) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/SumLong.java]: unused import

**Comment by hyunsik** [comment]
+1
Thank you for finding these bugs. It seems to correct wrong null handling behavior of aggregation operators.

**Comment by hyunsik** [comment]
Hi @sirpkt,

I leave some trivial comments. Because they are trivial, you can commit after you remove them.

**Comment by sirpkt** [comment]
I just committed the patch.

**Pull request closed without merging**